opensearch/src/main/java/org/opensearch/sql/opensearch/storage/script/filter/lucene/relevance/SimpleQueryString.java

```diff
@@ -0,0 +1,123 @@
+      (b, v) -> b.minimumShouldMatch(v.stringValue());
+  private final BiFunction<SimpleQueryStringBuilder, ExprValue, SimpleQueryStringBuilder> quoteFieldSuffix =
+      (b, v) -> b.quoteFieldSuffix(v.stringValue());
+  private final BiFunction<SimpleQueryStringBuilder, ExprValue, SimpleQueryStringBuilder> boost =
+      (b, v) -> b.boost(Float.parseFloat(v.stringValue()));
+
+  ImmutableMap<Object, Object> argAction = ImmutableMap.builder()
+      .put("analyze_wildcard", analyzeWildcard)
+      .put("analyzer", analyzer)
+      .put("auto_generate_synonyms_phrase_query", autoGenerateSynonymsPhraseQuery)
+      .put("flags", flags)
+      .put("fuzzy_max_expansions", fuzzyMaxExpansions)
+      .put("fuzzy_prefix_length", fuzzyPrefixLength)
+      .put("fuzzy_transpositions", fuzzyTranspositions)
+      .put("lenient", lenient)
+      .put("default_operator", defaultOperator)
+      .put("minimum_should_match", minimumShouldMatch)
+      .put("quote_field_suffix", quoteFieldSuffix)
+      .put("boost", boost)
+      .build();
+
+  @Override
+  public QueryBuilder build(FunctionExpression func) {
+    Iterator<Expression> iterator = func.getArguments().iterator();
+    NamedArgumentExpression fields = (NamedArgumentExpression) iterator.next();
+    NamedArgumentExpression query = (NamedArgumentExpression) iterator.next();
+    SimpleQueryStringBuilder queryBuilder = QueryBuilders.simpleQueryStringQuery(
+        query.getValue().valueOf(null).stringValue());
+    queryBuilder.fields(parseFields(fields.getValue().valueOf(null).stringValue()));
+    while (iterator.hasNext()) {
+      NamedArgumentExpression arg = (NamedArgumentExpression) iterator.next();
+      if (!argAction.containsKey(arg.getArgName())) {
+        throw new SemanticCheckException(String
+            .format("Parameter %s is invalid for simple_query_string function.", arg.getArgName()));
+      }
+      ((BiFunction<SimpleQueryStringBuilder, ExprValue, SimpleQueryStringBuilder>) argAction
+          .get(arg.getArgName()))
+          .apply(queryBuilder, arg.getValue().valueOf(null));
+    }
+    return queryBuilder;
+  }
+
+  private Map<String, Float> parseFields(String fields) {
+    try {
+      // TODO support elements wrapped by single quotes
+      var arr = new JSONArray(fields);
+      if (!arr.toList().stream().allMatch(s -> s instanceof String))
+        throw new Exception("All listed elements should be strings.");
+
+      var lst = arr.toList().stream().map(String::valueOf).collect(Collectors.toList());
+      var res = new HashMap<String, Float>();
+      //var builder = ImmutableMap.builder();
+      for (var elem : lst) {
+        if (!elem.contains("^")) {
+          res.put(elem, 1F);
+          continue;
+        }
+
+        var parts = elem.split("\\^");
+        var weight = Float.parseFloat(parts[parts.length - 1]);
+        var field = Arrays.stream(parts).limit(parts.length - 1).collect(Collectors.joining("^"));
```

Comment:
1. Should we support multiple `^` character?
2. The exception message would be unclear for the end user if string ends with `^` or there is no float after `^`.
3. `^` may be a part of the string, so it should be escaped.